File: clients/algoliasearch-client-php/lib/Model/Analytics/GetClickPositionsResponsePositions.php
<?php

namespace Algolia\AlgoliaSearch\Model\Analytics;

/**
 * GetClickPositionsResponsePositions Class Doc Comment
 *
 * @category Class
 * @package Algolia\AlgoliaSearch
 */
class GetClickPositionsResponsePositions extends \Algolia\AlgoliaSearch\Model\AbstractModel implements ModelInterface, \ArrayAccess, \JsonSerializable
{
    /**
      * Array of property to type mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelTypes = [
        'position' => 'int[]',
        'clickCount' => 'int',
    ];

    /**
      * Array of property to format mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelFormats = [
        'position' => null,
        'clickCount' => null,
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelTypes()
    {
        return self::$modelTypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelFormats()
    {
        return self::$modelFormats;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'position' => 'setPosition',
        'clickCount' => 'setClickCount',
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'position' => 'getPosition',
        'clickCount' => 'getClickCount',
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     */
    public function __construct(array $data = null)
    {
        if (isset($data['position'])) {
            $this->container['position'] = $data['position'];
        }
        if (isset($data['clickCount'])) {
            $this->container['clickCount'] = $data['clickCount'];
        }
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        if (!isset($this->container['position']) || $this->container['position'] === null) {
            $invalidProperties[] = "'position' can't be null";
        }
        if (!isset($this->container['clickCount']) || $this->container['clickCount'] === null) {
            $invalidProperties[] = "'clickCount' can't be null";
        }

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }

    /**
     * Gets position
     *
     * @return int[]
     */
    public function getPosition()
    {
        return $this->container['position'] ?? null;
    }

    /**
     * Sets position
     *
     * @param int[] $position Range of positions with the following pattern: Positions from 1 to 10 included are displayed in separated groups. Positions from 11 to 20 included are grouped together. Positions from 21 and up are grouped together.
     *
     * @return self
     */
    public function setPosition($position)
    {
        $this->container['position'] = $position;

        return $this;
    }

    /**
     * Gets clickCount
     *
     * @return int
     */
    public function getClickCount()
    {
        return $this->container['clickCount'] ?? null;
    }

    /**
     * Sets clickCount
     *
     * @param int $clickCount the number of click event
     *
     * @return self
     */
    public function setClickCount($clickCount)
    {
        $this->container['clickCount'] = $clickCount;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param int $offset Offset
     *
     * @return bool
     */
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param int $offset Offset
     *
     * @return mixed|null
     */
    public function offsetGet($offset)
    {
        return $this->container[$offset] ?? null;
    }

    /**
     * Sets value based on offset.
     *
     * @param int|null $offset Offset
     * @param mixed    $value  Value to be set
     *
     * @return void
     */
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param int $offset Offset
     *
     * @return void
     */
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }
}

File: clients/algoliasearch-client-java-2/algoliasearch-core/com/algolia/model/analytics/GetClickPositionsResponsePositions.java
package com.algolia.model.analytics;

import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

/** GetClickPositionsResponsePositions */
public class GetClickPositionsResponsePositions {

  @SerializedName("position")
  private List<Integer> position = new ArrayList<>();

  @SerializedName("clickCount")
  private Integer clickCount;

  public GetClickPositionsResponsePositions setPosition(
    List<Integer> position
  ) {
    this.position = position;
    return this;
  }

  public GetClickPositionsResponsePositions addPositionItem(
    Integer positionItem
  ) {
    this.position.add(positionItem);
    return this;
  }

  /**
   * Range of positions with the following pattern: Positions from 1 to 10 included are displayed in
   * separated groups. Positions from 11 to 20 included are grouped together. Positions from 21 and
   * up are grouped together.
   *
   * @return position
   */
  @javax.annotation.Nonnull
  public List<Integer> getPosition() {
    return position;
  }

  public GetClickPositionsResponsePositions setClickCount(Integer clickCount) {
    this.clickCount = clickCount;
    return this;
  }

  /**
   * The number of click event.
   *
   * @return clickCount
   */
  @javax.annotation.Nonnull
  public Integer getClickCount() {
    return clickCount;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    GetClickPositionsResponsePositions getClickPositionsResponsePositions = (GetClickPositionsResponsePositions) o;
    return (
      Objects.equals(
        this.position,
        getClickPositionsResponsePositions.position
      ) &&
      Objects.equals(
        this.clickCount,
        getClickPositionsResponsePositions.clickCount
      )
    );
  }

  @Override
  public int hashCode() {
    return Objects.hash(position, clickCount);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class GetClickPositionsResponsePositions {\n");
    sb.append("    position: ").append(toIndentedString(position)).append("\n");
    sb
      .append("    clickCount: ")
      .append(toIndentedString(clickCount))
      .append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
